### PR TITLE
Type fix

### LINKED ISSUE #2615: Typefix
🐛 Due to changes in variable types in the nightly PyTorch version, some # type ignore comments are redundant and we need some in some new places for mypy tests to pass Due to changes in variable types in the nightly PyTorch version, some # type ignore comments are redundant and we need some in some new places for mypy tests to pass.

 Mypy tests failing
 Install pytorch nightly version by `conda install pytorch torchvision torchaudio cudatoolkit=10.2 -c pytorch-nightly` and run `bash ./tests/run_code_style.sh mypy`
 
image
 Expected the tests to pass

## Environment

- PyTorch Version: 1.13.0.dev20220715
- Ignite Version: 0.5.0
- OS: Ubuntu 20.04
- How you installed Ignite: `pip`
- Python version: `3.10.4`

### PR DESCRIPTION
**Pull request by kamalojasv181**
Fixes #2615 

Description:  Deleted/Added the # type ignore comments wherever required. Also skipped mypy tests for the current stable version of PyTorch.

### CODE REVIEW

**Comment by kamalojasv181** [comment]
@vfdev-5 can you please approve the tests?

File: ignite/handlers/lr_finder.py [hunk 1]
```diff
@@ -493,9 +493,9 @@ def __init__(self, optimizer: Optimizer, start_lr: float, end_lr: float, num_ite
         super(_ExponentialLR, self).__init__(optimizer, last_epoch)
 
         # override base_lrs
-        self.base_lrs = start_lr
+        self.base_lrs = start_lr  # type: ignore
```
Comment (vfdev-5) [inline ignite/handlers/lr_finder.py:496]: As you already mentioned that in discord, let's try to explore why there is an error.

Comment (vfdev-5) [reply ignite/handlers/lr_finder.py:496]: Actually, the issue is that `self.base_lrs` is `List[float]` but `start_lr` is `float`. There is a type mismatch. I wonder if actually this `_ExponentialLR` is working correctly ...

Comment (vfdev-5) [reply ignite/handlers/lr_finder.py:496]: Oh, there is just a typehint to fix:

https://github.com/pytorch/ignite/blob/9402abeb67b35c2272ede7af4a2c64dd64aa7278/ignite/handlers/lr_finder.py#L125-L126

We can first fix typehint for this line:
```
start_lr = [start_lr] * len(optimizer.param_groups)  # type: ignore
```
and then update 
```
class _ExponentialLR(_LRScheduler):
    def __init__(self, optimizer: Optimizer, start_lr: float, end_lr: float, num_iter: int, last_epoch: int = -1):
```

Comment (kamalojasv181) [reply ignite/handlers/lr_finder.py:496]: Yes. Is the intended use what the description says?

Comment (vfdev-5) [reply ignite/handlers/lr_finder.py:496]: Intended usage is `start_lr: List[float]`

**Review by vfdev-5 — APPROVED** [review]
Thanks @kamalojasv181 !

**Pull request merged**